### DISCUSSION TITLE
Make use of new Chrome performance measurement APIs?

### DISCUSSION THREAD

**Discussion by danielcompton** [Day8/re-frame-10x#300] [Ideas]
I'm not sure what/how we might use these, but they might be worth looking at a bit later in the summer. https://developers.google.com/web/updates/2017/06/user-centric-performance-metrics

**Comment by danielcompton** [comment]
https://calendar.perfplanet.com/2017/tracking-cpu-with-long-tasks-api/
https://www.html5rocks.com/en/tutorials/webperformance/usertiming/

**Comment by mike-thompson-day8** [comment]
Better interaction with the [chrome profile API](https://developers.google.com/web/tools/chrome-devtools/console/console-reference#profile).  
See also `console.timeStamp([label])`

**Comment by mike-thompson-day8** [comment]
Closing — superseded by [day8/re-frame#840](https://github.com/day8/re-frame/issues/840) ("Bridge re-frame's trace timing into the User Timing API for in-DevTools profiling").

The 2017 framing here was "use Chrome's perf APIs" without a concrete shape. Eight years on, the right shape is clearer: re-frame's trace stream already carries high-resolution timing for every event/sub/fx/render — what's missing is publishing that timing into the W3C User Timing API so it interleaves with browser-native paint/layout/network/GC events on a single timeline in DevTools' Performance panel.

The new issue:

- Lives on re-frame (not 10x) — it's a generic capability for any re-frame consumer; 10x's role would just be a settings toggle that calls `(re-frame.devtools.perf/enable!)`.
- Names the structured label shape (`rf:event:cart/apply-coupon`, `rf:sub:cart/total`, `rf:render:re-com.box[cart/view.cljs:84]`, etc.) — including `{:file :line}` source coords from `re-frame.core-instrumented` when present.
- Lights up beyond Chrome — User Timing is a W3C standard; works in Firefox / Safari / Lighthouse / any `PerformanceObserver` consumer.
- Is opt-in (no overhead by default), with periodic flush so the marks list doesn't grow unbounded.

See [day8/re-frame#840](https://github.com/day8/re-frame/issues/840) for the full proposal.
